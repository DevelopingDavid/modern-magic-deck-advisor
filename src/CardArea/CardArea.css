.card-area {
  background-color: lightgray;
  height: 100vh;
}

.card-area--section {
  display: flex;
  flex-wrap: wrap;
}