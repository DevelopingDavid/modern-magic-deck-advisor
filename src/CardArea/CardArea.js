import React, { Component } from 'react';
import './CardArea.css';
import Card from '../Card/Card.js';
import Header from '../Header/Header.js';
import Deck from '../Deck/Deck.js';

class CardArea extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  sortCardsByName(card1, card2) {
    if (card1.cardName < card2.cardName) {
      return -1;
    } else {
      return 1;
    }
  }

  sortCards() {
    this.props.userCardsData.sort(this.sortCardsByName);
    return this.props.userCardsData.map(userCard => {
      return <Card userCard={userCard} key={userCard.cardName} />
    });
  }

  sortDecks() {
    this.props.userDecks.sort((a, b) => a.price - b.price);
    return this.props.userDecks.map(userDeck => {
      return <Deck userDeck={userDeck} key={userDeck.deckName} />
    });
  }
  
  displayPopup = (event) => {
    let cardPop = document.querySelector('.card--popup');
    if (event.target.classList.contains('card--image')) {
      cardPop.classList.add('card--pop');
      cardPop.classList.remove('card--popup-hide');
    }
  }

  returnToScreen = () => {
    let cardPop = document.querySelector('.card--popup');
    cardPop.classList.add('card--popup-hide');
    cardPop.classList.remove('card--pop');    
  }

  render(props) {
    let cardAreaView = [];
    if (this.props.cardAreaView === 'myCardList') {
      cardAreaView = this.sortCards();
    } else if (this.props.cardAreaView === 'compareDecks') {
      cardAreaView = this.sortDecks();
    }

    return (
      <div onClick={event => this.displayPopup(event)}
      className="card-area">
        <div className="card--popup  card--popup-hide">
          <img
            className="popup--card"
            src="https://img.scryfall.com/cards/large/en/a25/122.jpg"
          />
          <div className="popup-info">
          <div>
            <h1>Card Title</h1>
            <h1>Mana Cost:</h1>
            <h1>Card Type:</h1>
            <h1>Card Price:</h1>
            <h1>Card Avg:</h1>
          </div>
            <i onClick={this.returnToScreen} 
               className="far fa-times-circle"></i>
          </div>
        </div>
        <Header 
          setAsideView={this.props.setAsideView}/>
        <section className="card-area--section">
          {cardAreaView}
        </section>
      </div>
    );
  }
}

export default CardArea;
