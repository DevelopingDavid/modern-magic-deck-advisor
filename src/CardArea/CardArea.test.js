import React from 'react';
import CardArea from './CardArea';
import { shallow } from 'enzyme';
import cards from '../mockData';

const userCardsData = [
  { cardName: 'Blood Moon', cardCount: 2 }, 
  { cardName: 'Breeding Pool', cardCount: 2 },
  { cardName: 'Arid Mesa', cardCount: 2 }
];
const userDecks = [
  { deckName: 'Ponza' },
  { deckName: 'Burn' },
  { deckName: 'Hollow One' }
];
const faveDecks = [
  { deckName: "Bant Spirits" },
  { deckName: "Izzet Phoenix" }
];
const wishList = [
  { cardName: "Blood Moon"},
  { cardName: "Mox Opal" },
  { cardName: "Breeding Pool" }
];
const expandedDeckInfo = [
  { cardName: 'Blood Moon' }, 
  { cardName: 'Arid Mesa' }
]
const popUpView = false;
const cardAreaView = 'myCardList';
const setAsideViewMock = jest.fn(); 
const setCardAreaViewMock = jest.fn(); 
const setPopUpViewMock = jest.fn(); 
const getExpandedDeckInfoMock = jest.fn(() => expandedDeckInfo); 
const addToWishlistMock = jest.fn(); 
const addToFaveDecksMock = jest.fn(); 

describe('CardArea', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CardArea
        userCardsData={userCardsData}
        faveDecks={faveDecks}
        setAsideView={setAsideViewMock}
        cardAreaView={cardAreaView}
        setCardAreaView={setCardAreaViewMock}
        userDecks={userDecks}
        getExpandedDeckInfo={getExpandedDeckInfoMock}
        addToWishlist={addToWishlistMock}
        addToFaveDecks={addToFaveDecksMock}
        wishList={wishList}
        cards={cards}
        popUpView={popUpView}
        setPopUpView={setPopUpViewMock}/>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`should set popUpCard in state and call setPopUpView when
    displayPopUp is called`, () => {
      expect(wrapper.state('popUpCard').cardName).toEqual(undefined);
      wrapper.instance().displayPopUp(userCardsData[0]);
      expect(wrapper.state('popUpCard').cardName).toEqual('Blood Moon');
  });

  it('should call props.setPopUpView when returnToScreen is called', () => {
    wrapper.instance().returnToScreen();
    expect(setPopUpViewMock).toBeCalled();
  });

  it('should set cardsInDeck in state when expandDeck is called', () => {
    expect(wrapper.state('cardsInDeck').length).toEqual(0);
    wrapper.instance().expandDeck({
      deckName: 'some deck',
      cards: ['Blood Moon', 'Arid Mesa'],
      cardCounts: {
        'Blood Moon': 2,
        'Arid Mesa': 3
      }
    });
    expect(wrapper.state('cardsInDeck').length).toEqual(2);
  });

  it(`should return a string of a number with two decimal places
    when getPriceString is called`, () => {
      expect(wrapper.instance().getPriceString(12.345)).toEqual('12.35');
    });
});