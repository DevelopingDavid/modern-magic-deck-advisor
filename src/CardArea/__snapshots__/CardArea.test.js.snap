// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardArea should match the snapshot 1`] = `
<div
  className="card-area"
>
  <Header
    setAsideView={[MockFunction]}
  />
  <section
    className="card-area--section"
  >
    <Card
      card={
        Object {
          "cardName": "Blood Moon",
        }
      }
      cardAreaView="myCardList"
      displayPopUp={[Function]}
      key="Blood Moon"
    />
    <Card
      card={
        Object {
          "cardName": "Breeding Pool",
        }
      }
      cardAreaView="myCardList"
      displayPopUp={[Function]}
      key="Breeding Pool"
    />
    <Card
      card={
        Object {
          "cardName": "Arid Mesa",
        }
      }
      cardAreaView="myCardList"
      displayPopUp={[Function]}
      key="Arid Mesa"
    />
  </section>
</div>
`;
